src/lib/helper/arch_compat.h

```diff
@@ -0,0 +1,12 @@
+#pragma once
+
+#ifdef __SSE__
+#  include <immintrin.h>
+#  define PAUSE() _mm_pause()
+#else
+#  define PAUSE()
+#endif
+
+#ifdef __powerpc64__
+#  define __builtin_ia32_sfence()
```

Comment:
Maybe `atomic_thread_fence` would be the appropriate thing to use instead of the intrinsics? Not sure about the PPC execution model. Shouldn't keep us from pulling.